### PR TITLE
Issue #69,  #67 Refactored http repository, Added ReceiptMapping, Added NodeHttp

### PR DESCRIPTION
**Pull request by rg911**
- Refactored Http repos to match latest `Dragon` swagger definition
- Added `ReceiptMapping` to create receipt object from DTO
- Added `NodeHttp`

### CODE REVIEW

File: src/integration-test/java/io/nem/sdk/infrastructure/BlockHttpTest.java [hunk 1]
```diff
@@ -75,41 +76,18 @@ void getBlockTransactions() throws ExecutionException, InterruptedException {
...
-        assertTrue(blockchainStorageInfo.getNumTransactions() > 0);
-        assertTrue(blockchainStorageInfo.getNumBlocks() > 0);
-    }
-
     @Test
     void throwExceptionWhenBlockDoesNotExists() {
         TestObserver<BlockInfo> testObserver = new TestObserver<>();
-        blockchainHttp
+        blockHttp
                 .getBlockByHeight(BigInteger.valueOf(1000000000))
```
Comment (evias) [inline src/integration-test/java/io/nem/sdk/infrastructure/BlockHttpTest.java]: I would prefer "block 0" here. No matter how unrealistic it is that the chain goes up to this number of blocks, the number introduced there is definitely going to happen *some day*, making this test inconsistent.

File: src/main/java/io/nem/sdk/model/node/RoleType.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.nem.sdk.model.node;
+
+public enum RoleType {
+    PearNode(1),
```
Comment (evias) [inline src/main/java/io/nem/sdk/model/node/RoleType.java]: PeerNode typo.

File: src/main/java/io/nem/sdk/model/node/RoleType.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+    /**
+     * Static constructor converting role type raw value to enum instance.
+     *
+     * @return {@link RoleType}
+     */
+    public static RoleType rawValueOf(int value) {
+        switch (value) {
+            case 1:
+                return RoleType.PearNode;
```
Comment (evias) [inline src/main/java/io/nem/sdk/model/node/RoleType.java]: reproduce peernode typo..

File: src/integration-test/java/io/nem/sdk/infrastructure/BlockHttpTest.java (hunk 1 above)
Comment (rg911) [reply src/integration-test/java/io/nem/sdk/infrastructure/BlockHttpTest.java]: make sense, this was from copied from existing codes, didn't spotted it.

File: src/main/java/io/nem/sdk/model/node/RoleType.java (hunk 2 above)
Comment (evias) [reply src/main/java/io/nem/sdk/model/node/RoleType.java]: you like pears hu

**Pull request merged**